Kernel/Thread.h

```diff
@@ -126,7 +126,11 @@ class Thread {
     class Blocker {
     public:
         virtual ~Blocker() { }
-        virtual bool should_unblock(Thread&, time_t now_s, long us) = 0;
+        virtual bool should_unblock(Thread& thread, time_t, long)
+        {
+            return should_unblock(thread);
+        }
+        virtual bool should_unblock(Thread&) = 0;
```

Comment:
What's the point of breaking this into two separate virtuals?
A subclass would never want to override both.

Reply:
My vision for this is to remove the `virtual bool should_unblock(Thread& thread, time_t, long)` variant. Right now there are 3 or 4 that all implement the timeout logic (the same way), but the rest don't. For `Thread::block` I need to be able to check the block condition without taking a timeout into account, so I'm using `bool should_unblock(Thread&)` there. But I don't think the `Blocker` classes really should be in the business of determining whether a timeout happened, that in my mind should be universal and the same logic should apply to all.

So, for now, because of `Thread::block` I temporarily split them up, but my plan is to remove the first variant entirely.

Reply:
Okay, that's a good reason :)